OpenSim/Simulation/Model/Frame.h

```diff
@@ -189,9 +189,13 @@ OpenSim_DECLARE_ABSTRACT_OBJECT(Frame, ModelComponent);
     ///@}
 
     /** Add a Mesh specified by file name to the list of Geometry owned by the Frame.
-    Scale defaults to 1.0 but can be changed on the call line. For convenience 
+    Scale defaults to 1.0 but can be changed on the call line for convenience.
     */
-    void addMeshGeometry(const std::string &aGeometryFileName, const SimTK::Vec3 scale = SimTK::Vec3(1));
+    void attachMeshGeometry(const std::string &aGeometryFileName, const SimTK::Vec3 scale = SimTK::Vec3(1));
+    /** Add a piece of Geometry to the list of Geometry owned by the Frame.
+    Scale defaults to 1.0 but can be changed on the call line, for convenience.
+    */
+    void attachGeometry(OpenSim::Geometry& geom, const SimTK::Vec3 scale = SimTK::Vec3(1));
```

Comment:
Good point @chrisdembia, the problem is that the copying is done internally by the list property so I had to use a non-const reference to set values before copying (in Frame::attachGeometry). I also thought that making it non-const makes it clear that the method modifies the Geometry by setting the frame name and maybe scale. Do you think it'd be better to make it const, have the property create and add a copy then retrieve the copy and set the attributes on it?